LibreNMS/Modules/Routes.php

```diff
@@ -0,0 +1,363 @@
+<?php
+
+/**
+ * Route.php
+ *
+ * Route discovery module
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @link       http://librenms.org
+ *
+ * @copyright  2025 Peca Nesovanovic
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+ */
+
+namespace LibreNMS\Modules;
+
+use App\Facades\PortCache;
+use App\Models\Device;
+use App\Models\Route;
+use App\Observers\ModuleModelObserver;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Log;
+use LibreNMS\Config;
+use LibreNMS\DB\SyncsModels;
+use LibreNMS\Exceptions\InvalidIpException;
+use LibreNMS\Interfaces\Data\DataStorageInterface;
+use LibreNMS\Interfaces\Discovery\RouteDiscovery;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\OS;
+use LibreNMS\Polling\ModuleStatus;
+use LibreNMS\Util\IPv4;
+use LibreNMS\Util\IPv6;
+use SnmpQuery;
+
+class Routes implements Module
+{
+    use SyncsModels;
+
+    /**
+     * @inheritDoc
+     */
+    public function dependencies(): array
+    {
+        return ['ports'];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function shouldDiscover(OS $os, ModuleStatus $status): bool
+    {
+        return $status->isEnabledAndDeviceUp($os->getDevice());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function shouldPoll(OS $os, ModuleStatus $status): bool
+    {
+        return false;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function discover(OS $os): void
+    {
+        $update_timestamp = \Carbon\Carbon::now();
+        $max_routes = (null != Config::get('routes_max_number')) ? Config::get('routes_max_number') : 1000;
+
+        $routesFromDiscovery = new Collection;
+
+        if ($os instanceof RouteDiscovery) {
+            $routesFromDiscovery = $os->discoverRoutes();
+        }
+
+        if ($routesFromDiscovery->isEmpty()) {
```

Comment:
Hi @Npeca75 
I haven't tested it yet but it looks good! 
One point to ensure: 
- If `ipCidrRouteNumber` or `inetCidrRouteTable` return a value AND if this value is over `$max_route`, then no other call to any of the discovery functions without this check should be made. The goal is to try to protect discovery to kill the routers with Full Internet Table ... Currently, looking at the code below, all discover are tried one after the other.

Reply:
>but it looks good! 

tnx
 
>Currently, looking at the code below, all discover are tried one after the other.

ok, as you wish, done